include/alpaka/rand/Philox/PhiloxSingle.hpp

```diff
@@ -0,0 +1,138 @@
+/* Copyright 2021 Jiri Vyskocil
+ *
+ * This file is part of alpaka.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+#pragma once
+
+#include <alpaka/rand/Philox/MultiplyAndSplit64to32.hpp>
+#include <alpaka/rand/Philox/PhiloxBaseTraits.hpp>
+
+#include <utility>
+
+namespace alpaka
+{
+    namespace rand
+    {
+        namespace engine
+        {
+            /** Philox state for single value engine
+             *
+             * @tparam TCounter Type of the Counter array
+             * @tparam TKey Type of the Key array
+             */
+            template<typename TCounter, typename TKey>
+            struct PhiloxStateSingle
+            {
+                using Counter = TCounter;
+                using Key = TKey;
+
+                Counter counter; ///< Counter array
+                Key key; ///< Key array
+                Counter result; ///< Intermediate result array
+                std::uint32_t position; ///< Pointer to the active intermediate result element
+                // TODO: Box-Muller states
+            };
+
+            /** Philox engine generating a single number
+             *
+             * This engine's operator() will return a single number. Since the result is the same size as the counter,
+             * and so it contains more than one number, it has to be stored between individual invocations of
+             * operator(). Additionally a pointer has to be stored indicating which part of the result array is to be
+             * returned next.
```

Comment:
Is this really user facing documentation? This sounds more like notes for developers.

Reply:
This is the doxygen documentation. I was about to move this somewhere else, but then we talked about doxygen in the meeting and it seemed that these docs (where they exist at all) are targeted at the developers as well.